scarb-metadata/src/command.rs

```diff
@@ -112,36 +104,39 @@ impl MetadataCommand {
         K: AsRef<OsStr>,
         V: AsRef<OsStr>,
     {
-        for (ref key, ref val) in vars {
-            self.env(key, val);
-        }
+        self.inner.envs(vars);
         self
     }
 
     /// Removes an environment variable mapping.
-    pub fn env_remove<K>(&mut self, key: impl AsRef<OsStr>) -> &mut Self {
-        let key = key.as_ref();
-        if self.env_clear {
-            self.env.remove(key);
-        } else {
-            self.env.insert(key.to_os_string(), None);
-        }
+    pub fn env_remove(&mut self, key: impl AsRef<OsStr>) -> &mut Self {
+        self.inner.env_remove(key);
         self
     }
 
     /// Clears the entire environment map for the child process.
     pub fn env_clear(&mut self) -> &mut Self {
-        self.env.clear();
-        self.env_clear = true;
+        self.inner.env_clear();
         self
     }
 
     /// Inherit standard error, i.e. show Scarb errors in this process's standard error.
     pub fn inherit_stderr(&mut self) -> &mut Self {
-        self.inherit_stderr = true;
+        self.inner.inherit_stderr();
         self
     }
 
+    fn scarb_command(&self) -> Command {
+        let mut builder = self.inner.clone();
+        builder.json();
+        builder.args(["metadata", "--format-version"]);
+        builder.arg(VersionPin.numeric().to_string());
```

Comment:
This will break if somebody adds an arg that is specific to the `metadata` command. It will land before `metadata` command name, while it should be afterwards

Reply:
Discussed offline